components/workspace/hooks.tsx

```diff
@@ -95,6 +98,77 @@ export const useWorkspaceEmptyMutation = () => {
   return mutation
 }
 
+export const useWorkspaceExportMutation = () => {
+  const labelerAgent = useLabelerAgent()
+  return useMutation({
+    mutationFn: async (items: string[]) => {
+      // Items are exported in groups so we can expect all items in the group to be of same type
+      // For now, only support exporting accounts
+      if (!items[0].startsWith('did:')) {
+        toast.error(`Exporting is only enabled for accounts.`)
+        return []
+      }
+
+      const data: Record<
+        string,
+        ToolsOzoneModerationDefs.RepoViewDetail | null
+      > = {}
+
+      for (const itemChunk of chunkArray(items, 50)) {
+        await Promise.all(
+          itemChunk.map(async (did) => {
+            try {
+              const { data: repo } =
+                await labelerAgent.tools.ozone.moderation.getRepo({
+                  did,
+                })
+
+              data[did] = repo
+            } catch (error) {
+              // For now we're just swallowing errors and exporting what I can
+              console.error(error)
+              data[did] = null
+            }
+          }),
+        )
+      }
+
+      downloadCSV(
+        createCSV({
+          headers: [
+            'did',
+            'handle',
+            'email',
+            'ip',
+            'name',
+            'labels',
+            'profile',
+          ],
+          lines: Object.values(data).map((repo) => {
+            if (!repo) return ''
+            const profile = AppBskyActorProfile.isRecord(repo.relatedRecords[0])
+              ? (repo.relatedRecords[0] as AppBskyActorProfile.Record)
+              : null
+
+            const line: string[] = [
+              repo.did,
+              repo.handle,
+              repo.email || 'Unknown',
+              `${repo.ip || 'Unknown'}`,
+              `${profile?.displayName || 'Unknown'}`,
+              repo.labels?.map(({ val }) => val).join(', ') || 'None',
+              buildBlueSkyAppUrl({ did: repo.did }),
+            ]
+            return line.join(',')
```

Comment:
Values containing commas or newlines will end up breaking the CSV format.  If we add some escaping we should be able to address it, though:
```ts
const escapeCSVValue = (value: string) => {
  if (value.contains(',') || value.contains('"') || value.contains('\r') || value.contains('\n')) {
    return `"${value.replaceAll('"', '""')}"`
  }
  return value
}

// ...

  return line.map(escapeCSVValue).join(',')
```

Reply:
yh except for labels/tags list, we don't have anything that contain those characters though but will add to make it more future proof.